src/Resources/skeleton/model/Repository.tpl.php

```diff
@@ -0,0 +1,63 @@
+<?= "<?php\n" ?>
+
+namespace <?= $namespace; ?>;
+
+<?= $use_statements ?>
+
+class <?= $class_name ?> extends ServiceEntityRepository
+{
+    /**
+    * @extends ServiceEntityRepository<<?= $entity_class_name ?>>
+    *
+    * @method <?= $entity_class_name ?>|null find($id, $lockMode = null, $lockVersion = null)
+    * @method <?= $entity_class_name ?>|null findOneBy(array $criteria, array $orderBy = null)
+    * @method <?= $entity_class_name ?>[]    findAll()
+    * @method <?= $entity_class_name ?>[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+    */
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, <?= $entity_class_name ?>::class);
+    }
+
+    public function save(<?= $entity_class_name ?> $entity, bool $flush = false): void
+    {
+        $this->getEntityManager()->persist($entity);
+
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+    public function remove(<?= $entity_class_name ?> $entity, bool $flush = false): void
+    {
+        $this->getEntityManager()->remove($entity);
+
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+//    /**
+//    * @return <?= $entity_class_name ?>[] Returns an array of <?= $entity_class_name ?> objects
+//    */
+//    public function findByExampleField($value): array
+//    {
+//        return $this->filter(function(QueryBuilder $queryBuilder) use ($value) {
+//            $queryBuilder
+//                ->andWhere('t.exampleField = :val')
+//                ->setParamter('val', $value)
+//                ->orderBy('t.id', 'ASC')
+//            ;
+//        });
+//    }
+
+//    public function findOneBySomeField($value): ?<?= $entity_class_name ?><?= "\n" ?>
+//    {
+//        return $this->createQueryBuilder('t')
+//            ->andWhere('t.exampleField = :val')
+//            ->setParameter('val', $value)
+//            ->getQuery()
+//            ->getOneOrNullResult()
+//        ;
+//    }
+}
```

Comment:
**todo**: Okay, there some things mixed up, maybe I poorly explained 😄 

What you're creating here is a "standard" Doctrine repository.

The approach from [geekcell/ddd](https://github.com/geekcell/php-ddd) and this bundle is a bit different.
- A repository interface should live in `Domain/Repository` and extend [`GeekCell\Ddd\Contracts\Domain\Repository`](https://github.com/geekcell/php-ddd/blob/main/src/Contracts/Domain/Repository.php)
    - This interface contains the all methods we want to implement (like `findOneBySomeField(...)`)
- A class which implements this interface should live in `Infrastructure/Doctrine/ORM/Repository` and also extend [`GeekCell\DddBundle\Infrastructure\Doctrine\Repository`](https://github.com/geekcell/ddd-symfony-bundle/blob/main/src/Infrastructure/Doctrine/Repository.php)

Example

```php
// in src/Baz/Domain/Repository/FooRepository.php

use GeekCell\Ddd\Contracts\Domain\Repository;

interface FooRepository extends Repository
{
    public function findById(FooId $id): Foo;
    ...
}

// in src/Baz/Infrastructure/Doctrine/ORM/Repository/FooRepository.php

use Acme\Baz\Domain\Repository\FooRepository as FooRepositoryInterface;
use GeekCell\DddBundle\Infrastructure\Doctrine\Repository as OrmRepository;

class FooRepository extends OrmRepository implements FooRepositoryInterface
{
    ...
}
```

Doing it that way gives you access to `$this->filter(...)` and more.

You can find real world examples of this pattern in the sample repository you already know about.

Reply:
Changes made.